### PR TITLE
MGMT 1316 move install host to use new state machine model

### PR DESCRIPTION
**Pull request by filanov**

### CODE REVIEW

File: internal/host/transition.go [hunk 1]
```diff
@@ -143,3 +143,39 @@ func (th *transitionHandler) PostResetHost(sw stateswitch.StateSwitch, args stat
...
+
+func (th *transitionHandler) IsValidRoleForInstallation(sw stateswitch.StateSwitch, _ stateswitch.TransitionArgs) (bool, error) {
+	sHost, ok := sw.(*stateHost)
+	if !ok {
+		return false, errors.New("IsValidRoleForInstallation incompatible type of StateSwitch")
+	}
+	validRoles := []string{models.HostRoleMaster, models.HostRoleWorker}
+	if !funk.ContainsString(validRoles, sHost.host.Role) {
+		return false, common.NewApiError(http.StatusConflict,
+			errors.Errorf("Can't install host %s doe invalid host role %s should be one of %s",
```
Comment (eranco74) [inline internal/host/transition.go:164]: typo
```suggestion
			errors.Errorf("Can't install host %s due to invalid host role: %s, should be one of %s",
```

File: internal/host/transition.go [hunk 2]
```diff
@@ -143,3 +143,39 @@ func (th *transitionHandler) PostResetHost(sw stateswitch.StateSwitch, args stat
...
+
+func (th *transitionHandler) IsValidRoleForInstallation(sw stateswitch.StateSwitch, _ stateswitch.TransitionArgs) (bool, error) {
+	sHost, ok := sw.(*stateHost)
+	if !ok {
+		return false, errors.New("IsValidRoleForInstallation incompatible type of StateSwitch")
+	}
+	validRoles := []string{models.HostRoleMaster, models.HostRoleWorker}
+	if !funk.ContainsString(validRoles, sHost.host.Role) {
+		return false, common.NewApiError(http.StatusConflict,
+			errors.Errorf("Can't install host %s doe to invalid host role: %s, should be one of %s",
```
Comment (ori-amizur) [inline internal/host/transition.go:164]: Typo doe

File: internal/host/transition.go [hunk 3]
```diff
@@ -143,3 +143,39 @@ func (th *transitionHandler) PostResetHost(sw stateswitch.StateSwitch, args stat
...
+				sHost.host.ID.String(), sHost.host.Role, validRoles))
+	}
+	return true, nil
```
Comment (ori-amizur) [inline internal/host/transition.go:167]: This boolean function always returns true or error.  It is more like validation function.  If it wouldn't return error but false, then it an be used also for the negative case that it is not valid (maybe move to error state)

File: internal/host/statemachine.go [hunk 4]
```diff
@@ -26,33 +27,53 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
...
+		SourceStates:     []stateswitch.State{HostStatusKnown},
+		DestinationState: HostStatusInstalling,
+		PostTransition:   th.PostInstallHost,
+	})
+
+	// Install disabled host will not do anything
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType:   TransitionTypeInstallHost,
+		SourceStates:     []stateswitch.State{HostStatusDisabled},
+		DestinationState: HostStatusDisabled,
```
Comment (ori-amizur) [inline internal/host/statemachine.go:75]: Why do we need this transition?

File: internal/host/transition.go (hunk 2 above)
Comment (filanov) [reply internal/host/transition.go:164]: will be fixed in other PR

File: internal/host/transition.go (hunk 3 above)
Comment (filanov) [reply internal/host/transition.go:167]: It's an easy way to return specific http error

File: internal/host/statemachine.go (hunk 4 above)
Comment (filanov) [reply internal/host/statemachine.go:75]: Disable is a special case, we don't want to ignore hosts in disabled state while trying to install. so we don't want to change the state and don't want to return an error. 
Any other suggestion to do it? (there is a comment above the transition)

Comment (ori-amizur) [reply internal/host/statemachine.go:75]: Will there be an error if no transition rule is fired?

File: internal/host/transition.go (hunk 3 above)
Comment (ori-amizur) [reply internal/host/transition.go:167]: Do we assume that it will be always be called from HTTP request?

Comment (filanov) [reply internal/host/transition.go:167]: Yes

File: internal/host/statemachine.go (hunk 4 above)
Comment (filanov) [reply internal/host/statemachine.go:75]: Yes

**Pull request merged**